spark/src/main/scala/org/apache/spark/filodb/FiloSetup.scala

```diff
@@ -106,20 +119,47 @@ object FiloDriver extends FiloSetup with StrictLogging {
 object FiloExecutor extends FiloSetup with StrictLogging {
   /**
    * Initializes the config if it is not set, and start things for an executor.
-   * @param filoConfig The config within the filodb.** level.
+    *
+    * @param filoConfig The config within the filodb.** level.
    * @param role the Akka Cluster role, either "executor" or "driver"
    */
   def init(filoConfig: Config): Unit = synchronized {
+    var startIngestion = true
     _config.getOrElse {
       this.role = "executor"
       _config = Some(filoConfig)
-      kamonInit()
+        kamonInit()
+
       coordinatorActor       // force coordinator to start
-      // get address from config and join cluster.  note: it's ok to join cluster multiple times
-      val addr = AddressFromURIString.parse(filoConfig.getString("spark-driver-addr"))
-      logger.info(s"Initializing FiloExecutor clustering by joining driver at $addr...")
-      cluster.join(addr)
-      Thread sleep 1000
+
+      // start reloading WAL files
+      implicit val timeout = Timeout(FiniteDuration(10, SECONDS))
+      val resp = coordinatorActor ? ReloadDCA
```

Comment:
Let's move all this code out to a separate function.  It will make merging later changes much, much easier.